CMakeLists.txt

```diff
@@ -280,8 +280,8 @@ if (USE_GNUTLS)
 	endif()
 	target_compile_definitions(datachannel PRIVATE USE_GNUTLS=1)
 	target_compile_definitions(datachannel-static PRIVATE USE_GNUTLS=1)
-	target_link_libraries(datachannel PRIVATE GnuTLS::GnuTLS)
-	target_link_libraries(datachannel-static PRIVATE GnuTLS::GnuTLS)
+	target_link_libraries(datachannel PUBLIC GnuTLS::GnuTLS)
+	target_link_libraries(datachannel-static PUBLIC GnuTLS::GnuTLS)
```

Comment:
Why is that change necessary? GnuTLS, OpenSSL and libnice dependencies should be linked against but not exposed to dependencies for linking, hence the private visibility.

Reply:
I thinks GnuTLS, OpenSSL and libnice is same Threads. The application need link they.

Reply:
Actually, the application does not need to include GnuTLS headers directories, and only libdatachannel.so needs to link against libgnutls.so, the application does not.

Reply:
Threads is a bit different as not only libdatachannel needs to be built with thread flags, the application also needs to. That's why Threads is public.